tests/components/azure_event_hub/test_init.py

```diff
@@ -90,16 +76,94 @@ async def test_full_config(hass):
             },
         }
     }
-    config[azure_event_hub.DOMAIN].update(MIN_CONFIG)
     assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
 
 
+async def test_setup(hass, mock_hub):
+    """Test the async_setup function."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await azure_event_hub.async_setup_entry(hass, entry)
+    assert hass.data[azure_event_hub.DOMAIN]["hub"] is not None
```

Comment:
Please remove all hass.data assertions.

Reply:
I removed all, except in the options flow tests, where I do an assert to see if the update has been done.